pkg/omf/omf.fish

```diff
@@ -136,11 +138,14 @@ function omf -d "Oh My Fish"
 
     case "u" "update"
       echo (omf::em)"Updating Oh My Fish..."(omf::off)
-      if omf.repo.pull $OMF_PATH
-        echo (omf::em)"Oh My Fish is up to date."(omf::off)
-      else
-        echo (omf::err)"Oh My Fish failed to update."(omf::off)
-        echo "Please open a new issue here → "(omf::em)"github.com/oh-my-fish/oh-my-fish/issues"(omf::off)
+      switch omf.repo.pull $OMF_PATH
+        case $OMF_UNKNOWN_ERR
+          echo (omf::dim)"Oh My Fish already up to date!"(omf::off)
+        case $OMF_SUCCESS
+          echo (omf::em)"Oh My Fish updated!"(omf::off)
+        case '*'
+          echo (omf::err)"Oh My Fish failed to update."(omf::off)
+          echo "Please open a new issue here → "(omf::em)"github.com/oh-my-fish/oh-my-fish/issues"(omf::off)
```

Comment:
With this in mind, we need a way to disambiguate three things that can happen when pulling a repository:

- Changes detected. Package updated.
- No remote changes. Package not updated.
- Error while updating. Package not updated.

In this switch/case block the following approach was taken:

- `OMF_UNKNOWN_ERROR`: Nothing updated
- `OMF_SUCCESS`: Something updated
- `*`: If any other error, failed update

We have some possibilities for `omf.repo.pull` return codes:

- Define specific return codes: `OMF_REPO_UPDATED`, for example
- Use the `-1`, `0` and `+1` for not updated, updated and errored
- Move error message logic into `omf.repo.pull`

I would like some comments on the matter to continue development, thanks!